pkg/converter/pod_converter.go

```diff
@@ -60,26 +56,23 @@ func NewPodConverter() Converter {
 }
 
 func (p *podConverter) Convert(k8sObj interface{}) (interface{}, error) {
-	if reflect.TypeOf(k8sObj) != reflect.TypeOf(&k8sApiV1.Pod{}) {
-		log.Fatalf("can not convert object %#v to workloadEndpoint. Object is not of type *v1.Pod", k8sObj)
-	}
-
+	// Convert Pod into a workload endpoint.
+	var c k8s.Converter
+	var bc backendConverter.WorkloadEndpointConverter
 	pod := k8sObj.(*k8sApiV1.Pod)
-	endpoint := api.NewWorkloadEndpoint()
-
-	endpoint.Metadata.Workload = fmt.Sprintf("%s.%s", pod.Namespace, pod.Name)
-	if pod.ObjectMeta.Labels != nil {
-		endpoint.Metadata.Labels = pod.ObjectMeta.Labels
-	} else {
-		endpoint.Metadata.Labels = map[string]string{}
+	kvp, err := c.PodToWorkloadEndpoint(pod)
+	if err != nil {
+		return nil, err
```

Comment:
Just noting `NewPodController()` handles error w/ an error log rather than current Fatal behavior. (this seems fine).

Reply:
Yeah I think that's the right behavior.